org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/extensions/relaxng/utils/RelaxNGUtils.java

```diff
@@ -38,4 +68,307 @@ private static String getRelaxNGURI(RelaxNGGrammar grammar) {
 		return grammar.getGrammarDescription().getExpandedSystemId();
 	}
 
+	/**
+	 * Binding type of rng grammar

```

Comment:
Is this enum needed? It seems like DEFINE is used to signal that an element is `ref` with a `name` attribute. Are there other binding types that you think you may add? Otherwise, maybe it's better to rename `getBindingType` to `isRefBoundToName` and make it boolean.

Reply:
I would to keep this enum to be consistent with XSD and perhaps there is another binding type.

Reply:
Okay. Let's keep the enum then. I think you should improve the javadoc to better describe what the enum represents.

Reply:
I add a comment.